python/lsst/dm/rucio/register/rucio_interface.py

```diff
@@ -0,0 +1,249 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import hashlib
+import logging
+import random
+import time
+import zlib
+
+import lsst.daf.butler
+import rucio.common.exception
+from lsst.dm.rucio.register.resource_bundle import ResourceBundle
+from rucio.client.didclient import DIDClient
+from rucio.client.replicaclient import ReplicaClient
+
+__all__ = ["RucioInterface"]
+
+logger = logging.getLogger(__name__)
+
+
+class RucioInterface:
+    """Add files as replicas in Rucio, along with metadata,
+    and attach them to datasets.
+
+    Parameters
+    ----------
+    butler: `lsst.daf.butler.Butler`
+        Butler we're operating upon
+    rucio_rse: `str`
+        Name of the RSE that the files live in.
+    scope: `str`
+        Rucio scope to register the files in.
+    rse_root: `str`
+        Full path to root directory of RSE directory structure
+    dtn_url: `str`
+        Base URL of the data transfer node for the Rucio physical filename.
+    """
+
+    def __init__(
+        self,
+        butler: lsst.daf.butler.Butler,
+        rucio_rse: str,
+        scope: str,
+        rse_root: str,
+        dtn_url: str,
+    ):
+        self.butler = butler
+        self.rse = rucio_rse
+        self.scope = scope
+        self.rse_root = rse_root
+        self.dtn_url = dtn_url
+        self.pfn_base = f"{dtn_url}"
+
+    def _make_bundle(self, dataset_id, dataset_ref) -> ResourceBundle:
+        """Make a ResourceBundle
+
+        Parameters
+        ----------
+        dataset_id: `str`
+            Rucio dataset name
+        dataset_ref: `DatasetRef`
+            Butler DatasetRef
+        """
+        did = self._make_did(self.butler.getURI(dataset_ref), dataset_ref.to_json())
+        rb = ResourceBundle(dataset_id, did)
+        return rb
+
+    def _make_did(self, resource_path, metadata: str) -> dict[str, str | int]:
+        """Make a Rucio data identifier dictionary from a resource.
+
+        Parameters
+        ----------
+        resource_path: ResourcePath
+            ResourcePath object
+
+        metadata: `str`
+            String containing Rubin dataset specific metadata
+
+        Returns
+        -------
+        did: `dict [ str, str|int ]`
+            Rucio data identifier including physical and logical names,
+            byte length, adler32 and MD5 checksums, meta, and scope.
+        """
+        logging.info(f"type(resource_path) = {type(resource_path)}")
```

Comment:
This doesn't seem like an info level message since it's going to return HttpResourcePath or FileResourcePath or S3ResourcePath and not the actual URI with the embedded scheme.

Reply:
That wasn't meant to be included.  I removed it.